nengo/processes.py

```diff
@@ -208,6 +213,13 @@ def make_step(self, shape_in, shape_out, dt, rng):
         coefficients *= np.sqrt(2 * n_coefficients)
         signal = np.fft.irfft(coefficients, axis=0)
 
+        if self.y0 is not None:
+            # Starts each dimension off where it is closest to y0
+            def shift(x):
+                offset = np.argmin(abs(self.y0 - x))
+                return np.roll(x, -offset+1)  # +1 since t starts at dt
+            signal = np.apply_along_axis(shift, 0, signal)
```

Comment:
Note: `roll` here takes an axis argument, but it doesn't help prevent the `apply_along_axis` because `offset` could be different per dimension in `shape[1:]`.